Proxy service panics when you try to join via an existing proxy tunnel

Expected behavior:
Works?

Current behavior:
```
panic: runtime error: index out of range [0] with length 0

goroutine 60 [running]:
github.com/gravitational/teleport/lib/service.(*TeleportProcess).initProxyEndpoint(0xc000fee680, 0xc000d4e918)
	github.com/gravitational/teleport/lib/service/service.go:3670 +0x6906
github.com/gravitational/teleport/lib/service.(*TeleportProcess).initProxy.func1()
	github.com/gravitational/teleport/lib/service/service.go:2995 +0x65
github.com/gravitational/teleport/lib/service.(*LocalService).Serve(0xc000fe1180?)
	github.com/gravitational/teleport/lib/service/supervisor.go:539 +0x1a
github.com/gravitational/teleport/lib/service.(*LocalSupervisor).serve.func1()
	github.com/gravitational/teleport/lib/service/supervisor.go:276 +0x175
created by github.com/gravitational/teleport/lib/service.(*LocalSupervisor).serve
	github.com/gravitational/teleport/lib/service/supervisor.go:271 +0xaa
```


Bug details:

- Teleport version
 Teleport v12.2.3 git:api/v12.2.3-0-gea25c87 go1.20.3
 
- Recreation steps
I am trying to run a cluster with 2 proxies, first teleport is running with the following config:
```
version: v2
teleport:
  nodename: teleport
  data_dir: /var/lib/teleport
  log:
    output: stderr
    severity: INFO
    format:
      output: text
  ca_pin: ""
  diag_addr: ""
auth_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3025
  cluster_name: teleport-test.myproject.com
  proxy_listener_mode: multiplex
  session_recording: "proxy"
ssh_service:
  enabled: false
proxy_service:
  enabled: "yes"
  web_listen_addr: 0.0.0.0:443
  public_addr: teleport-test-private.myproject.com:443
  https_keypairs:
  - key_file: /etc/letsencrypt/live/teleport-test.myproject.com/privkey.pem
    cert_file: /etc/letsencrypt/live/teleport-test.myproject.com/fullchain.pem
```

Config of the second proxy that breaks:
```
version: v3
teleport:
  nodename: teleport
  data_dir: /var/lib/teleport-public-proxy
  proxy_server: teleport-test-private.myproject.com:443
  auth_token: /home/ubuntu/teleport-public-proxy-token.file
auth_service:
  enabled: false
ssh_service:
  enabled: false
proxy_service:
  enabled: true
  web_listen_addr: 127.0.0.1:3000
  public_addr: teleport-test.myproject.com:3000
  https_keypairs:
  - key_file: /etc/letsencrypt/live/teleport-test.myproject.com/privkey.pem
    cert_file: /etc/letsencrypt/live/teleport-test.myproject.com/fullchain.pem
```

- Debug logs
```
ubuntu@teleport-test:~$ sudo teleport start --config /etc/teleport-public-proxy.yaml --debug
2023-04-14T18:00:48Z INFO             Starting Teleport v12.2.3 with a config file located at "/etc/teleport-public-proxy.yaml" common/teleport.go:522
2023-04-14T18:00:48Z DEBU [SQLITE]    Connected to: file:%2Fvar%2Flib%2Fteleport-public-proxy%2Fproc%2Fsqlite.db?_busy_timeout=10000&_sync=FULL&_txlock=immediate, poll stream period: 1s lite/lite.go:246
2023-04-14T18:00:48Z DEBU [SQLITE]    journal_mode=delete, synchronous=2, busy_timeout=10000 lite/lite.go:293
2023-04-14T18:00:49Z DEBU [PROC:1]    Adding service to supervisor. pid:48405.1 service:register.proxy service/supervisor.go:214
2023-04-14T18:00:49Z DEBU [PROC:1]    Adding service to supervisor. pid:48405.1 service:proxy.init service/supervisor.go:214
2023-04-14T18:00:49Z DEBU [PROC:1]    Adding service to supervisor. pid:48405.1 service:common.rotate service/supervisor.go:214
2023-04-14T18:00:49Z DEBU [PROC:1]    Adding service to supervisor. pid:48405.1 service:common.upload service/supervisor.go:214
2023-04-14T18:00:49Z DEBU [PROC:1]    Adding service to supervisor. pid:48405.1 service:register.instance service/supervisor.go:214
2023-04-14T18:00:49Z DEBU [PROC:1]    Adding service to supervisor. pid:48405.1 service:instance.init service/supervisor.go:214
2023-04-14T18:00:49Z DEBU [PROC:1]    No signal pipe to import, must be first Teleport process. pid:48405.1 service/service.go:1167
2023-04-14T18:00:49Z DEBU [PROC:1]    Service has started. pid:48405.1 service:register.proxy service/supervisor.go:275
2023-04-14T18:00:49Z DEBU [PROC:1]    Connected state: never updated. pid:48405.1 service/connect.go:239
2023-04-14T18:00:49Z INFO [PROC:1]    Connecting to the cluster teleport-test.myproject.com with TLS client certificate. pid:48405.1 service/connect.go:258
2023-04-14T18:00:49Z DEBU [PROC:1]    Attempting to connect to Auth Server through tunnel. pid:48405.1 proxy-server:teleport-test-private.myproject.com:443 service/connect.go:1135
2023-04-14T18:00:49Z DEBU [PROC:1]    Service has started. pid:48405.1 service:proxy.init service/supervisor.go:275
2023-04-14T18:00:49Z DEBU [PROC:1]    Service has started. pid:48405.1 service:common.rotate service/supervisor.go:275
2023-04-14T18:00:49Z DEBU [PROC:1]    Service has started. pid:48405.1 service:common.upload service/supervisor.go:275
2023-04-14T18:00:49Z DEBU [PROC:1]    Service has started. pid:48405.1 service:register.instance service/supervisor.go:275
2023-04-14T18:00:49Z DEBU [PROC:1]    Connected state: never updated. pid:48405.1 service/connect.go:239
2023-04-14T18:00:49Z INFO [PROC:1]    Connecting to the cluster teleport-test.myproject.com with TLS client certificate. pid:48405.1 service/connect.go:258
2023-04-14T18:00:49Z DEBU [PROC:1]    Attempting to connect to Auth Server through tunnel. pid:48405.1 proxy-server:teleport-test-private.myproject.com:443 service/connect.go:1135
2023-04-14T18:00:49Z DEBU [PROC:1]    Service has started. pid:48405.1 service:instance.init service/supervisor.go:275
2023-04-14T18:00:49Z DEBU             Attempting GET teleport-test-private.myproject.com:443/webapi/find webclient/webclient.go:129
2023-04-14T18:00:49Z DEBU             Attempting GET teleport-test-private.myproject.com:443/webapi/find webclient/webclient.go:129
2023-04-14T18:00:49Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-04-14T18:00:49Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-04-14T18:00:49Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-04-14T18:00:49Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-04-14T18:00:49Z DEBU [PROC:1]    Connected to Auth Server through tunnel. pid:48405.1 proxy-server:teleport-test-private.myproject.com:443 service/connect.go:1142
2023-04-14T18:00:49Z DEBU [PROC:1]    Connected client: Identity(Instance, cert(5de7ca2b-8d32-485a-bf96-4b564c25570d.teleport-test.myproject.com issued by teleport-test.myproject.com:258069983453864879188000185027398719202),trust root(teleport-test.myproject.com:258069983453864879188000185027398719202)) pid:48405.1 service/connect.go:204
2023-04-14T18:00:49Z DEBU [PROC:1]    Connected to Auth Server through tunnel. pid:48405.1 proxy-server:teleport-test-private.myproject.com:443 service/connect.go:1142
2023-04-14T18:00:49Z DEBU [PROC:1]    Connected client: Identity(Proxy, cert(5de7ca2b-8d32-485a-bf96-4b564c25570d.teleport-test.myproject.com issued by teleport-test.myproject.com:258069983453864879188000185027398719202),trust root(teleport-test.myproject.com:258069983453864879188000185027398719202)) pid:48405.1 service/connect.go:204
2023-04-14T18:00:49Z INFO [PROC:1]    Proxy: features loaded from auth server: Kubernetes:true App:true DB:true Desktop:true  pid:48405.1 service/connect.go:92
2023-04-14T18:00:49Z DEBU [PROC:1]    Broadcasting event. event:ProxyIdentity pid:48405.1 service/supervisor.go:383
2023-04-14T18:00:49Z DEBU [PROC:1]    Service is completed and removed. pid:48405.1 service:register.proxy service/supervisor.go:252
2023-04-14T18:00:49Z DEBU [PROC:1]    Received event "ProxyIdentity". pid:48405.1 service/service.go:529
2023-04-14T18:00:49Z DEBU [PROC:1]    Creating in-memory backend for [proxy]. pid:48405.1 service/service.go:1923
2023-04-14T18:00:49Z INFO [PROC:1]    Instance: features loaded from auth server: Kubernetes:true App:true DB:true Desktop:true  pid:48405.1 service/connect.go:92
2023-04-14T18:00:49Z DEBU [PROC:1]    Broadcasting event. event:InstanceIdentity pid:48405.1 service/supervisor.go:383
2023-04-14T18:00:49Z DEBU [PROC:1]    Service is completed and removed. pid:48405.1 service:register.instance service/supervisor.go:252
2023-04-14T18:00:49Z DEBU [INSTANCE:] Received event "InstanceIdentity". pid:48405.1 service/service.go:529
2023-04-14T18:00:49Z INFO [INSTANCE:] Successfully registered instance client. pid:48405.1 service/service.go:2178
2023-04-14T18:00:49Z DEBU [PROC:1]    Broadcasting event. event:InstanceReady pid:48405.1 service/supervisor.go:383
2023-04-14T18:00:49Z DEBU [PROC:1]    Service is completed and removed. pid:48405.1 service:instance.init service/supervisor.go:252
2023-04-14T18:00:49Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-04-14T18:00:49Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-04-14T18:00:49Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-04-14T18:00:49Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-04-14T18:00:49Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-04-14T18:00:49Z INFO [PROXY:1:C] Cache "proxy" first init succeeded. cache/cache.go:889
2023-04-14T18:00:49Z DEBU [PROC:1]    Setup Proxy: Web Proxy Address: 127.0.0.1:3000, Reverse Tunnel Proxy Address:  pid:48405.1 service/service.go:3115
2023-04-14T18:00:49Z DEBU [PROC:1]    Setup Proxy: Proxy protocol is enabled for web service, multiplexing is on. pid:48405.1 service/service.go:3249
2023-04-14T18:00:49Z INFO [PROC:1]    Service proxy:web is creating new listener on 127.0.0.1:3000. pid:48405.1 service/signals.go:214
2023-04-14T18:00:49Z DEBU [SSH:PROXY] Supported ciphers: ["aes128-gcm@openssh.com" "chacha20-poly1305@openssh.com" "aes128-ctr" "aes192-ctr" "aes256-ctr"]. pid:48405.1 sshutils/server.go:300
2023-04-14T18:00:49Z DEBU [SSH:PROXY] Supported KEX algorithms: ["curve25519-sha256" "curve25519-sha256@libssh.org" "ecdh-sha2-nistp256" "ecdh-sha2-nistp384" "ecdh-sha2-nistp521" "diffie-hellman-group14-sha256"]. pid:48405.1 sshutils/server.go:310
2023-04-14T18:00:49Z DEBU [SSH:PROXY] Supported MAC algorithms: ["hmac-sha2-256-etm@openssh.com" "hmac-sha2-256"]. pid:48405.1 sshutils/server.go:320
2023-04-14T18:00:49Z DEBU [PROC:1]    Adding service to supervisor. pid:48405.1 service:proxy.reversetunnel.server service/supervisor.go:214
2023-04-14T18:00:49Z DEBU [PROXY:CER] Loading TLS certificate /etc/letsencrypt/live/teleport-test.myproject.com/fullchain.pem and key /etc/letsencrypt/live/teleport-test.myproject.com/privkey.pem. service/certreloader.go:105
2023-04-14T18:00:49Z DEBU [PROC:1]    Service is completed and removed. pid:48405.1 service:proxy.init service/supervisor.go:252
panic: runtime error: index out of range [0] with length 0

goroutine 30 [running]:
github.com/gravitational/teleport/lib/service.(*TeleportProcess).initProxyEndpoint(0xc0003cc1a0, 0xc00063b320)
  github.com/gravitational/teleport/lib/service/service.go:3670 +0x6906
github.com/gravitational/teleport/lib/service.(*TeleportProcess).initProxy.func1()
  github.com/gravitational/teleport/lib/service/service.go:2995 +0x65
github.com/gravitational/teleport/lib/service.(*LocalService).Serve(0xc00039f030?)
  github.com/gravitational/teleport/lib/service/supervisor.go:539 +0x1a
github.com/gravitational/teleport/lib/service.(*LocalSupervisor).serve.func1()
  github.com/gravitational/teleport/lib/service/supervisor.go:276 +0x175
created by github.com/gravitational/teleport/lib/service.(*LocalSupervisor).serve
  github.com/gravitational/teleport/lib/service/supervisor.go:271 +0xaa
```

--- a/lib/config/configuration.go
+++ b/lib/config/configuration.go
@@ -522,6 +522,10 @@ func applyAuthOrProxyAddress(fc *FileConfig, cfg *servicecfg.Config) error {
 		}
 
 		if haveProxyServer {
+			if fc.Proxy.Enabled() {
+				return trace.BadParameter("proxy_server can not be specified when proxy service is enabled")
+			}
+
 			addr, err := utils.ParseHostPortAddr(fc.ProxyServer, defaults.HTTPListenPort)
 			if err != nil {
 				return trace.Wrap(err)

--- a/lib/config/configuration_test.go
+++ b/lib/config/configuration_test.go
@@ -1138,6 +1138,17 @@ func TestProxyPeeringPublicAddr(t *testing.T) {
 	}
 }
 
+func TestProxyMustJoinViaAuth(t *testing.T) {
+	cfg := servicecfg.MakeDefaultConfig()
+
+	err := ApplyFileConfig(&FileConfig{
+		Version: defaults.TeleportConfigVersionV3,
+		Proxy:   Proxy{Service: Service{EnabledFlag: "yes"}},
+		Global:  Global{ProxyServer: "proxy.example.com:3080"},
+	}, cfg)
+	require.True(t, trace.IsBadParameter(err), "expected bad parameter, got %v", err)
+}
+
 func TestBackendDefaults(t *testing.T) {
 	read := func(val string) *servicecfg.Config {
 		// Default value is lite backend.

